internal/services/onelakeshortcut/schema.go

```diff
@@ -0,0 +1,664 @@
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	superschema "github.com/orange-cloudavenue/terraform-plugin-framework-superschema"
+
+	"github.com/microsoft/terraform-provider-fabric/internal/framework/customtypes"
+	"github.com/microsoft/terraform-provider-fabric/internal/pkg/fabricitem"
+)
+
+func itemSchema(isList bool) superschema.Schema { //revive:disable-line:flag-parameter
+	var dsTimeout *superschema.DatasourceTimeoutAttribute
+
+	if !isList {
+		dsTimeout = &superschema.DatasourceTimeoutAttribute{
+			Read: true,
+		}
+	}
+
+	return superschema.Schema{
+		Resource: superschema.SchemaDetails{
+			MarkdownDescription: fabricitem.NewResourceMarkdownDescription(ItemTypeInfo, false),
+		},
+		DataSource: superschema.SchemaDetails{
+			MarkdownDescription: fabricitem.NewDataSourceMarkdownDescription(ItemTypeInfo, isList),
+		},
+		Attributes: map[string]superschema.Attribute{
+			"id": superschema.StringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The " + ItemTypeInfo.Name + " ID.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Computed: true,
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"name": superschema.StringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The " + ItemTypeInfo.Name + " name.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+
+					Validators: []validator.String{
+						stringvalidator.LengthAtMost(200),
+						stringvalidator.LengthAtLeast(1),
+					},
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.RequiresReplace(),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Optional: !isList,
+					Computed: true,
+				},
+			},
+			"workspace_id": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The Workspace ID.",
+					CustomType:          customtypes.UUIDType{},
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: !isList,
+					Computed: isList,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.RequiresReplace(),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Required: !isList,
+					Computed: isList,
+				},
+			},
+			"item_id": superschema.SuperStringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "Item ID.",
+					CustomType:          customtypes.UUIDType{},
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+					Computed: isList,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.RequiresReplace(),
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Required: !isList,
+					Computed: isList,
+				},
+			},
+			"path": superschema.StringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "A string representing the full path where the shortcut is created, including either \"Files\" or \"Tables\".",
```

Comment:
nit - the syntax can be more readable.

```suggestion
					MarkdownDescription: `A string representing the full path where the shortcut is created, including either "Files" or "Tables".`,
```